dev/fattest.simplicity/src/componenttest/rules/repeater/FeatureSet.java

```diff
@@ -0,0 +1,122 @@
+ * Copyright (c) 2020 IBM Corporation and others.
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ *
+ * Contributors:
+ *     IBM Corporation - initial API and implementation
+ *******************************************************************************/
+package componenttest.rules.repeater;
+
+import java.util.Collections;
+import java.util.HashSet;
+import java.util.Set;
+
+/**
+ * An immutable set of features with an ID
+ */
+public class FeatureSet {
+    private final String id;
+    private final Set<String> features;
+
+    /**
+     * Create a new FeatureSet with the given ID and set of features
+     *
+     * @param id       The ID of the FeatureSet. It is recommended that this should be unique.
+     * @param features The features to include in the set
+     */
+    public FeatureSet(String id, Set<String> features) {
+        this.id = id;
+        this.features = Collections.unmodifiableSet(features);
+    }
+
+    /**
+     * Get the ID of this FeatureSet
+     *
+     * @return the id
+     */
+    public final String getID() {
+        return this.id;
+    }
+
+    /**
+     * Get an unmodifiable set of the features
+     *
+     * @return the features
+     */
+    public final Set<String> getFeatures() {
+        return this.features;
+    }
+
+    /**
+     * Create a new FeatureSetBuilder based on this FeatureSet
+     *
+     * @param  feature the feature to add to the set
+     * @return         a FeatureSetBuilder
+     */
+    public FeatureSetBuilder addFeature(String feature) {
+        FeatureSetBuilder builder = new FeatureSetBuilder(this);
+        builder.addFeature(feature);
+        return builder;
+    }
+
+    /**
+     * Create a new FeatureSetBuilder based on this FeatureSet
+     *
+     * @param  feature the feature to remove from the set
+     * @return         a FeatureSetBuilder
+     */
+    public FeatureSetBuilder removeFeature(String feature) {
+        FeatureSetBuilder builder = new FeatureSetBuilder(this);
+        builder.removeFeature(feature);
+        return builder;
+    }
+
+    public static class FeatureSetBuilder {
+
+        private final HashSet<String> features;
+
+        /**
+         * @param featureSet
+         */
+        public FeatureSetBuilder(FeatureSet featureSet) {
+            this(featureSet.getFeatures());
+        }
+
+        /**
+         * @param features
+         */
+        public FeatureSetBuilder(Set<String> features) {
+            this.features = new HashSet<>(features);
+        }
+
+        /**
+         *
+         */
+        public FeatureSetBuilder() {
+            this.features = new HashSet<>();
+        }
+
+        /**
+         * @param feature
+         */
+        public FeatureSetBuilder addFeature(String feature) {
```

Comment:
Ideally these methods would have Javadoc too, though I think the class is fairly intuitive. Otherwise I'd take the tags out completely.